functions/__fzf_search_current_dir.fish

```diff
@@ -11,13 +11,13 @@ function __fzf_search_current_dir --description "Search the current directory. R
 
     # If the current token is a directory and has a trailing slash,
     # then use it as fd's base directory.
-    if string match --quiet -- "*/" $token && test -d $expanded_token
+    if string match --quiet -- "*/" $expanded_token && test -d $expanded_token
         set --append fd_arguments --base-directory=$expanded_token
         # use the directory name as fzf's prompt to indicate the search is limited to that directory
-        set --append fzf_arguments --prompt=$token --preview="__fzf_preview_file $token{}"
-        set file_paths_selected $token(fd $fd_arguments 2>/dev/null | fzf $fzf_arguments)
+        set --append fzf_arguments --prompt=$expanded_token --preview="__fzf_preview_file $expanded_token{}"
+        set file_paths_selected $expanded_token(fd $fd_arguments 2>/dev/null | fzf $fzf_arguments)
     else
-        set --append fzf_arguments --query=$token --preview='__fzf_preview_file {}'
+        set --append fzf_arguments --query=$expanded_token --preview='__fzf_preview_file {}'
```

Comment:
I don't see the bug. We used `token` intentionally because in all of these instances, ~ does expand to $HOME

Reply:
Without those if I do a search with `~/` as token and select `desktop` I end up with `'./~/desktop'`.

The result doesn't work because it's escaped.

Reply:
Did some more tests, the `expanded_token` is only needed there